gpii/node_modules/userListeners/test/userFolderTests.js

```diff
@@ -0,0 +1,290 @@
+        directory: "",
+        filename: "",
+        expect: "reject"
+    }
+]);
+
+/**
+ * Get an instance of the gpii.userListeners.userFolder component.
+ * @return {Component} The gpii.userListeners.userFolder component.
+ */
+gpii.tests.userListener.getUserFolderListener = function () {
+    var userListeners = gpii.userListeners({
+        listeners: {
+            "onCreate.startListeners": "fluid.identity"
+        },
+        events: {
+            "onListenersStart": null
+        },
+        distributeOptions: {
+            userFolder: {
+                record: "gpii.tests.userListener.userFolderListener",
+                target: "{that userFolder}.options.gradeNames"
+            }
+        }
+    });
+
+    return userListeners.userFolder;
+};
+
+jqUnit.asyncTest("userFolder readTokenFile", function () {
+
+    var tests = gpii.tests.userListener.tokenFileTests;
+
+    jqUnit.expect(tests.length * 2);
+    var tempDir = path.join(os.tmpdir(), "gpii-listener-tests" + Math.random());
+    fs.mkdirSync(tempDir);
+    var tempFiles = [];
+
+    teardowns.push(function () {
+        while (tempFiles.length) {
+            fs.unlinkSync(tempFiles.pop());
+        }
+        fs.rmdirSync(tempDir);
+    });
+
+    var doTest = function (index) {
+        if (index >= tests.length) {
+            jqUnit.start();
+        } else {
+            var currentTest = tests[index];
+            var messagePrefix = " - test index " + index;
+
+            if (currentTest.hasOwnProperty("content")) {
+                var tempFile = path.join(tempDir, currentTest.filename);
+                fs.writeFileSync(tempFile, currentTest.content, "utf8");
+                tempFiles.push(tempFile);
+            }
+
+            var directory = currentTest.hasOwnProperty("directory") ? currentTest.directory : tempDir;
+            var file = currentTest.filename;
+
+            var p = gpii.userListeners.readTokenFile(directory, file);
+
+            jqUnit.assertTrue("readTokenFile must return a promise" + messagePrefix, fluid.isPromise(p));
+
+            p.then(function (token) {
+                jqUnit.assertEquals("readTokenFile should resolve with the expected value" + messagePrefix,
+                    currentTest.expect, token);
+
+                doTest(index + 1);
+            }, function () {
+                jqUnit.assertEquals("readTokenFile should reject if expected" + messagePrefix,
+                    currentTest.expect, "reject");
+
+                doTest(index + 1);
+            });
+        }
+    };
+
+    doTest(0);
+});
+
+jqUnit.asyncTest("userFolder writeTokenFile", function () {
+
+    var tempDir = os.tmpdir();
+    var tokenFile = "test-token-file" + Math.random();
+    var tokenPath = path.join(tempDir, tokenFile);
+
+    teardowns.push(function () {
+        try {
+            fs.unlinkSync(tokenPath);
+        } catch (e) {
+            // ignore
+        }
+    });
+
+    fluid.promise.sequence([
+        function () {
```

Comment:
The functions for three test cases in this sequence are pretty much identical. It would be good if you can consolidate them into one. Thanks.